users/views.py

```diff
@@ -1,3 +1,69 @@
-from django.shortcuts import render
+import json, bcrypt, jwt
+
+from django.conf  import settings
+from django.views import View
+from django.http  import JsonResponse
+from django.forms import ValidationError
+
+from .models    import User
+from .validator import password_validator
+
+class SignUpView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+            user         = data['user']
+            password     = data['password']
+            address      = data['address']
+            phone_number = data['phone_number']
+
+            password_validator(password)
+            
+            if User.objects.filter(user=user).exists():
+                return JsonResponse({'message':'ID_ALREADY_EXISTS'}, status=400)
+            
+            if User.objects.filter(phone_number=phone_number).exists():
+                return JsonResponse({'message':'PHONE_NUMBER_ALREADY_EXISTS'}, status=400)
+
+            hashed_password = bcrypt.hashpw(password.encode('utf-8'),bcrypt.gensalt()).decode('utf-8')
```

Comment:
여기서 `utf-8`로 decode해서 저장하는 이유는 무엇인가요?
```suggestion
            hashed_password = bcrypt.hashpw(password.encode('utf-8'),bcrypt.gensalt()).decode('utf-8')
```

Reply:
db에 들어갈 때 str 타입으로 저장되므로 bytes 타입으로 들어가게 되면 이후 checkpw 작업을 할 때 추가로 encode 해줘야 하고, 이때 ValueError: Invalid salt 에러가 나옵니다